pkg/validator/validators.go

```diff
@@ -186,6 +189,47 @@ func (kc *policyValidator) ingressBackendValidator(req *admissionv1.AdmissionReq
 	return nil, nil
 }
 
+// retryValidator validates the Retry custom resource
+func retryValidator(req *admissionv1.AdmissionRequest) (*admissionv1.AdmissionResponse, error) {
```

Comment:
If the source and destination of a retry policy are the same as an existing retry policy with a different name should it be rejected?

Reply:
You're right! But will hold off on this PR (put into draft mode), to work on "Retries: consolidate retries into UpstreamTrafficSetting API" #4799 for now.